lib/SILGen/SILGenDistributed.cpp

```diff
@@ -244,12 +384,27 @@ void SILGenFunction::emitDistributedActorReady(
   // Only designated initializers get the lifecycle handling injected
   assert(ctor->isDesignatedInit());
 
-  SILValue transport = findFirstDistributedActorSystemArg(F);
+  auto &C = ctor->getASTContext();
+  auto *dc = ctor->getDeclContext();
+  auto classDecl = dc->getSelfClassDecl();
 
   FullExpr scope(Cleanups, CleanupLocation(loc));
   auto borrowedSelf = actorSelf.borrow(*this, loc);
 
-  emitActorReadyCall(B, loc, borrowedSelf.getValue(), transport);
+  // --- load the actor system from the actor instance
+  ManagedValue actorSystem;
+  SGFContext sgfCxt; // TODO: is this right?
```

Comment:
AFAIK (could be totally wrong on this...) this is probably right -- I've seen it in other places where just an empty one was used like that

Reply:
ah ok thanks. I spotted that somewhere too and it seemed to work so 🤷🏼